sqld/src/replication/primary/logger.rs

```diff
@@ -885,9 +887,11 @@ impl ReplicationLogger {
     }
 }
 
-fn checkpoint_db(data_path: &Path) -> anyhow::Result<()> {
+fn checkpoint_db(data_path: &Path, disable_auto_checkpoint: bool) -> anyhow::Result<()> {
     unsafe {
         let conn = rusqlite::Connection::open(data_path)?;
+        conn.query_row("PRAGMA journal_mode=WAL", (), |_| Ok(()))?;
```

Comment:
Ok, I think there's a substantial issue here -- we cannot just create a new vanilla connection to the db, we need to create one with our WAL hook, that has overridden `on_checkpoint` behavior: https://github.com/libsql/sqld/blob/bb27f190a58400f29c43c8419126bfa0bf6d4429/sqld/src/replication/primary/logger.rs#L171 . Only then we'll properly perform a checkpoint that runs our bottomless code, right?